test/unit/calculators/checks_benefits_support_calculator_test.rb

```diff
@@ -221,473 +335,709 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "no"
+            assert_not calculator.eligible_for_tax_free_childcare?
+          end
 
-          calculator.where_do_you_live = "england"
-          calculator.children_living_with_you = "yes"
-          calculator.age_of_children = "1_or_under,3_to_4"
-          assert_not calculator.eligible_for_free_childcare_2yr_olds?
+          should "be false if working over 16 hours with a disabled child and children aged 18 to 19" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "yes"
+            calculator.children_with_disability = "yes"
+            calculator.age_of_children = "18_to_19"
+            assert_not calculator.eligible_for_tax_free_childcare?
+          end
         end
       end
 
-      context "#eligible_for_childcare_3_4yr_olds_wales??" do
-        should "return true if eligible for Childcare 3 and 4 Year Olds Wales" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.where_do_you_live = "wales"
-          calculator.are_you_working = "yes_over_16_hours_per_week"
-          calculator.children_living_with_you = "yes"
-          calculator.age_of_children = "3_to_4"
-          assert calculator.eligible_for_childcare_3_4yr_olds_wales?
+      context "#eligible_for_free_childcare_2yr_olds?" do
+        context "when eligible" do
+          should "be true if country is not Scotland, with a child aged 2" do
+            %w[england wales northern-ireland].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.children_living_with_you = "yes"
+              calculator.age_of_children = "2"
+              assert calculator.eligible_for_free_childcare_2yr_olds?
+            end
+          end
         end
 
-        should "return false if not eligible for Childcare 3 and 4 Year Olds Wales" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.where_do_you_live = "scotland"
-          assert_not calculator.eligible_for_childcare_3_4yr_olds_wales?
-
-          calculator.where_do_you_live = "wales"
-          calculator.are_you_working = "no"
-          assert_not calculator.eligible_for_childcare_3_4yr_olds_wales?
-
-          calculator.where_do_you_live = "wales"
-          calculator.are_you_working = "yes_under_16_hours_per_week"
-          calculator.children_living_with_you = "yes"
-          calculator.age_of_children = "3_to_4"
-          assert_not calculator.eligible_for_childcare_3_4yr_olds_wales?
+        context "when ineligible" do
+          should "be false if country is not Scotland with a child that is not 2" do
```

Comment:
should this test be testing that the child IS 2?

Reply:
no, the truthful test is
`Not Scotland, Child is 2`
so we need to test as false
`Not Scotland, Child not 2`
`Scotland, Child is 2`

so this is correct!